library/objective-c/EnvoyEngineImpl.m

```diff
@@ -66,40 +62,51 @@ static envoy_headers toNativeHeaders(EnvoyHeaders *headers) {
   return (envoy_headers){length, header_array};
 }
 
+static const void* ios_http_filter_init(const void *context) {
+  EnvoyHTTPFilterFactory *filterFactory = (__bridge EnvoyHTTPFilterFactory *)context;
+  EnvoyHTTPFilter *filter = filterFactory.create();
+  return CFBridgingRetain(filter);
+}
+
 static envoy_filter_headers_status
-ios_http_filter_on_request_headers(envoy_headers headers, bool end_stream, void *context) {
+ios_http_filter_on_request_headers(envoy_headers headers, bool end_stream, const void *context) {
   // TODO(goaway): optimize unmodified case
-  ios_http_filter_context *c = (ios_http_filter_context *)context;
-  if (c->filter.onRequestHeaders == nil) {
+  EnvoyHTTPFilter *filter = (__bridge EnvoyHTTPFilter *)context;
+  if (filter.onRequestHeaders == nil) {
```

Comment:
how come the pointer doen't need dereference here?

Reply:
Yeah, seems weird if you've been looking at C++ for a while. Technically speaking we're using shorthand to send an Objective-C message to `filter` - the expanded syntax would be `[filter onRequestHeaders]`. This just happens to be what the shorthand looks like.